README.md

```diff
@@ -1,35 +1,251 @@
 
 [![Codacy Badge](https://api.codacy.com/project/badge/grade/bc3a79d1b12649158a1eb4758e872141)](https://www.codacy.com/app/Codacy/codacy-engine-scala-seed)
+[![Codacy Badge](https://api.codacy.com/project/badge/Coverage/bc3a79d1b12649158a1eb4758e872141)](https://www.codacy.com/app/Codacy/codacy-engine-scala-seed?utm_source=github.com&utm_medium=referral&utm_content=codacy/codacy-engine-scala-seed&utm_campaign=Badge_Coverage)
 [![Build Status](https://circleci.com/gh/codacy/codacy-engine-scala-seed.svg?style=shield&circle-token=:circle-token)](https://circleci.com/gh/codacy/codacy-engine-scala-seed)
 [![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.codacy/codacy-engine-scala-seed_2.12/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.codacy/codacy-engine-scala-seed_2.12)
 
-We use external tools at Codacy, this is the library we use across the multiple external tools integrations.
-For more details and examples of tools that use this project, you can check
-[PMDJava](https://github.com/codacy/codacy-pmdjava),
-[ESLint](https://github.com/codacy/codacy-eslint) and
-[Pylint](https://github.com/codacy/codacy-pylint).
+Framework to help integration with external analysis tools at Codacy.
+
+For more details and examples of tools that use this project, you can check:
+* [PMDJava](https://github.com/codacy/codacy-pmdjava)
+* [ESLint](https://github.com/codacy/codacy-eslint)
+* [Pylint](https://github.com/codacy/codacy-pylint)
 
 ## Usage
 
 Add to your SBT dependencies:
 
 ```scala
-"com.codacy" % "codacy-engine-scala-seed_2.12" % "<VERSION>"
+"com.codacy" %% "codacy-engine-scala-seed" % "<VERSION>"
 ```
 
-You shouldn't worry about the library itself, we use it as a core in our tools,
-and everything is well explained in our Docs section.
-
 ## Docs
 
-[Docker Docs](http://docs.codacy.com/v1.5/docs/tool-developer-guide)
+### How to integrate an external analysis tool on Codacy
+
+By creating a docker and writing code to handle the tool invocation and output,
+you can integrate the tool of your choice on Codacy!
+
+> To know more about dockers, and how to write a docker file please refer to [https://docs.docker.com/reference/builder/](https://docs.docker.com/reference/builder/)
+
+In this tutorial, we explain how you can integrate an analysis tool of your choice in Codacy.
+You can check the code of an already implemented tool and, if you wish, fork it to start your own.
+You are free to modify and use it for your own tools.
+
+#### Requirements
+
+* Docker definition with the tool you want to integrate
+* Define the documentation for the patterns provided by the tool
+
+#### Assumptions and Behaviour
+
+* To run the tool we provide the configuration file, `/.codacyrc`, with the language to run and optional parameters your tool might need.
+* The source code to be analysed will be located in `/src`, meaning that when provided in the configuration, the file paths are relative to `/src`.
+
+* **.codacyrc**
```

Comment:
I'd change this line to something like "Structure of the codacyrc file:", so it's obvious what the list is about.